### DISCUSSION TITLE
Copilot copy-pasting

### DISCUSSION THREAD

**Discussion by skycatminepokie** [community/community#15195] [Copilot Conversations]
I've only been using Copilot for an hour, and I've already noticed it copy-pasting from other repositories, including showing the author name and tag for some classes. I'm already worried about copilot. I won't be using copilot for a while (even though it is helpful) because it can quite obviously plagiarize other's code. Hopefully the model will become more open-minded and start thinking of its own solutions instead of taking too strong of a liking to other's code. I know that it's difficult, but seeing real, personal names is already worrying me.

**Comment by FilipKufalov** [comment]
It depends on your pre-defined variables since the Copilot uses other Github repos as learning so if you have a blank project It will probably just grab some Github project, but if you have variables first it will copy the code according to your variables, I encounter this only on empty/not started projects but later as progress goes it suggests only codes according to my environment

**Reply by skycatminepokie** [reply]
That makes sense. However, this is a somewhat sizeable project (not huge, but not empty). Made a new Java class and it suggested pretty much the whole thing at once, but it had someone's real name in it (found out they were on GitHub, but there were 9 people under the same name). It's just kinda freaky. I don't wanna have to worry about violating copyright law when I'm just writing code :/

**Reply by FilipKufalov** [reply]
I asked Github Support about Copyright law situation they confirmed that codes published using Copilot are saved, that was my first question when I apply for a whitelist since I don't want to test something that can get me in trouble for reason like that
